tendermint/src/application.rs

```diff
@@ -37,7 +38,7 @@ use crate::types::{
 /// application.
 ///
 /// [`Server`]: crate::Server
-pub trait ABCIApplication: Send + Clone + 'static {
+pub trait ABCIApplication<G>: Send + Clone + 'static {
```

Comment:
It's a question. 
Why do we need this generic in declaration?

It seems like the `gaia-rs` won't compile after the change.